h2o-algos/src/main/java/hex/modelselection/ModelSelectionUtils.java

```diff
@@ -153,6 +188,325 @@ public static Frame[] generateMaxRTrainingFrames(ModelSelectionModel.ModelSelect
+                    final int resCount = modelCount++;
+                    final int[] subsetIndices = allPreds.clone();
+                    resA[resCount] = new RecursiveAction() {
+                        @Override
+                        protected void compute() {
+                            int lastSweepIndex = prevCPM.length-1;
+                            double[][] subsetCPM = addNewPred2CPM(allCPM, prevCPM, subsetIndices, pred2CPMIndices, 
+                                    hasIntercept);
+                            int lastPredInd = subsetIndices[subsetIndices.length-1];
+                            int newPredCPMLength = pred2CPMIndices[lastPredInd].length;
+                            if (newPredCPMLength == 1) {
+                                subsetMSE[resCount] = applySweep2LastPred(sweepVec, subsetCPM, newPredCPMLength);
+                            } else {
+                                SweepVector[][] newSweepVec = mapBasicVector2Multiple(sweepVec, newPredCPMLength);
+                                subsetMSE[resCount] = applySweep2LastPred(newSweepVec, subsetCPM, newPredCPMLength);
+                            }
+                            // apply new sweeps due to the addition of the new rows/columns
+                            int lastSubsetIndex = subsetCPM.length-1;
+                            int[] sweepIndices = IntStream.range(0, newPredCPMLength).map(x->x+lastSweepIndex).toArray();
+                            sweepCPM(subsetCPM, sweepIndices, false);
+                            subsetMSE[resCount] = subsetCPM[lastSubsetIndex][lastSubsetIndex];
+                        }
+                    };
+                }
+            } else {    // start from first predictor
+                final int resCount = modelCount++;
+                final int[] subsetIndices = allPreds.clone();
+                allSubsetList.add(IntStream.of( allPreds ).boxed().toArray( Integer[]::new ));
+                resA[resCount] = new RecursiveAction() {
+                    @Override
+                    protected void compute() {
+                       // generate CPM corresponding to the subset indices in subsetIndices
+                        double[][] subsetCPM = extractPredSubsetsCPM(allCPM, subsetIndices, pred2CPMIndices, hasIntercept);
+                        int lastSubsetIndex = subsetCPM.length-1;
+                       // perform sweeping action and record the sweeping vector and save the changed cpm
+                        sweepCPM(subsetCPM, IntStream.range(0, lastSubsetIndex).toArray(), false);
+                       // copy over the CPM after sweeping and sweeping vector to main program
+                        subsetMSE[resCount] = subsetCPM[lastSubsetIndex][lastSubsetIndex];
+                    }
+                };
+            }
+        }
+        Arrays.stream(resA).filter(x -> x != null).forEach(ForkJoinTask::invoke);
```

Comment:
I don't know the proper solution but I wonder if the usage of `RecursiveAction` where it does not seem to be doing any recursion is appropriate. Also `invoke`[1] seems to be blocking so the run is likely to be sequential. If that's the intention, simple anonymous function might be clearer to understand.

[1] https://github.com/h2oai/h2o-3/blob/d86ef62817af4faba2fae6b07351724ac6d3bf19/h2o-core/src/main/java/jsr166y/ForkJoinTask.java#L88-L89

Reply:
I treat this just as a multi-thread operation.  I don't know that if I invoke only one, it will be sequential.  That will really defeat the purpose of what I am trying to do here.

Reply:
@tomasfryda I use the parallel stream here and that should unblock things.  I used the RecursiveAction by copying what Tomas N has done before.  If you prefer another one, please let me know and I can put it in.  Thanks.  W

Reply:
@wendycwong If I understand the documentation properly the `invoke` is blocking since it should be "semantically equivalent to `fork(); join()`".  So I think it does work by offloading one task to another thread and then waiting until it finishes the computation and then it runs the second task in the same way and so on. 

I believe using `invokeAll` should run it in parallel:
```java
 ForkJoinTask.invokeAll(Arrays.stream(resA).filter(x -> x != null).toArray())
```

Reply:
@wendycwong I like the use of parallel stream and it should resolve the issue, however, I've been told not to use them and instead use h2o's own parallel tools. IIRC parallel stream would use threads outside of h2o's scheduler and that could be bad.
Unfortunately, I don't know much about that so maybe @michalkurka will say that the parallel stream is ok in this case since the threads are just waiting for the threads that are doing the actual work.

Reply:
@wendycwong Tomas suggested this:

    ForkJoinTask.invokeAll(Arrays.stream(resA).filter(x -> x != null).toArray())

This should take care of the null issue and it is the h2o-way (uses our f/j pool and should play nicer in h2o application). Is there an issue with his suggestion?

Reply:
I have problem using this:  

ForkJoinTask.invokeAll(Arrays.stream(resA).filter(x -> x != null).toArray())

<img width="719" alt="image" src="https://user-images.githubusercontent.com/7231712/175076652-351671bc-66ab-4979-a01b-a873e83bad18.png">

I used the suggestion and got the following run error:

<img width="1072" alt="image" src="https://user-images.githubusercontent.com/7231712/175076916-89f39fd0-24c1-480b-9e91-f14bd06bb063.png">

Tried a different cast and still got error:

<img width="1132" alt="image" src="https://user-images.githubusercontent.com/7231712/175077144-fa2e7f27-43e6-45f4-bc29-1c6a265731f5.png">


I was only able to pass test if I use my original version:

 Arrays.stream(resA).parallel().filter(x->x!=null).forEach(ForkJoinTask::invoke);

I need help!!!

Reply:
Using the generator argument in the `toArray` works for me:
```java
ForkJoinTask.invokeAll(Arrays.stream(resA).filter(Objects::nonNull).toArray(RecursiveAction[]::new));
```
(
Judging from the following tests: <img width="697" alt="Screen Shot 2022-06-23 at 16 48 49" src="https://user-images.githubusercontent.com/61695433/175328602-3eb420cb-fffa-484e-a91d-67e314a1c17b.png">
)

Reply:
Thanks TomasF, I changed it.